components/list/list-item-drag-handle.js

```diff
@@ -0,0 +1,196 @@
+
+import '../button/button-icon.js';
+import '../icons/icon.js';
+import { css, html, LitElement } from 'lit-element/lit-element.js';
+import { buttonStyles } from '../button/button-styles.js';
+import { getFirstFocusableDescendant } from '../../helpers/focus.js';
+
+const keyCodes = Object.freeze({
+	END: 35,
+	HOME: 36,
+	UP: 38,
+	DOWN: 40,
+	SPACE: 32,
+	ENTER: 13,
+	ESC: 27,
+	TAB: 9,
+	LEFT: 37,
+	RIGHT: 39,
+
+});
+
+export const dragActions = Object.freeze({
+	first: 'last',
+	end: 'first',
+	up: 'up',
+	down: 'down',
+	active: 'keyboard-active',
+	save: 'keyboard-deactivate-save',
+	cancel: 'keyboard-deactivate-cancel',
+	nextElement: 'next-element',
+	previousElement: 'previous-element'
+});
+
+class listItemDragHandle extends LitElement {
+
+	static get properties() {
+		return {
+			keyboardActive: { type: Boolean, attribute: 'keyboard-active', reflect: true },
+			text: { type: String }
+		};
+	}
+
+	static get styles() {
+		return [ buttonStyles, css`
+			:host {
+				align-items: center;
+				display: flex;
+				margin: 0.25rem;
+			}
+			:host([hidden]) {
+				display: none;
+			}
+			.d2l-list-item-drag-handle-dragger-button,
+			.d2l-list-item-drag-handle-keyboard-button {
+				display: grid;
+				grid-auto-columns: auto;
+				grid-auto-rows: 1fr 1fr;
+				margin: 0;
+				min-height: 1.8rem;
+				padding: 0;
+				width: 0.9rem;
+			}
+			/* Firefox includes a hidden border which messes up button dimensions */
+			button::-moz-focus-inner {
+				border: 0;
+			}
+			.d2l-list-item-drag-handle-dragger-button {
+				background-color: unset;
+			}
+			button,
+			button[disabled]:hover,
+			button[disabled]:focus {
+				background-color: var(--d2l-color-gypsum);
+				color: var(--d2l-color-ferrite);
+			}
+			button:hover,
+			button:focus {
+				background-color: var(--d2l-color-mica);
+			}
+			button[disabled] {
+				cursor: default;
+				opacity: 0.5;
+			}
+		`];
+	}
+
+	constructor() {
+		super();
+		this.keyboardActive = false;
+	}
+
+	focus() {
+		const node = getFirstFocusableDescendant(this);
+		if (node) {
```

Comment:
Micro-nit: in the olden days we always formatted statements like this, but in the past year or so we've relaxed that a bit in the interest of more succinct code.  Feel free to go `if (node) node.focus()` if you prefer.

Reply:
I did notice you start doing that - I'll change that as I prefer the shorter in some cases.